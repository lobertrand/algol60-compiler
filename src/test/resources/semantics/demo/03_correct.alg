begin
    integer procedure getMessageNumber(grade);
    value grade; integer grade;
    begin
        comment If-statement exhaustiveness check;
        if grade < 8 then
            getMessageNumber := 1
        else if grade < 14 then
            getMessageNumber := 2
        else
            getMessageNumber := 3
    end;

    integer a;
    a := 1;
    begin
        string array messages[a:3];
        string msg;

        messages[1] := "bad";
        messages[2] := "okay";
        messages[3] := "great";

        msg := messages[getMessageNumber(12)];
    end
end
