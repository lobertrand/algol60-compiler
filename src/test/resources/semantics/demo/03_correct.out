Algol60> Lexical analysis successful
Algol60> Syntactic analysis successful
┌───────────────────────────────────────────────┐
│ level: 0, tableNumber: 0                      │
├───────────────────────────────────────────────┤
│ Procedure: void outstring(integer, string)    │
│ Procedure: void outboolean(integer, boolean)  │
│ Procedure: void outinteger(integer, integer)  │
│ Procedure: void outreal(integer, real)        │
│ Procedure: void inreal(integer, real)         │
│ Procedure: void ininteger(integer, integer)   │
│ Procedure: integer entier(real)               │
└───────────────────────────────────────────────┘
    ┌───────────────────────────────────────────────┐
    │ level: 1, tableNumber: 1                      │
    ├───────────────────────────────────────────────┤
    │ Procedure: integer getMessageNumber(integer)  │
    │ Variable: integer a                           │
    └───────────────────────────────────────────────┘
        ┌───────────────────────────────────────────────────┐
        │ level: 2, tableNumber: 2                          │
        ├───────────────────────────────────────────────────┤
        │ Variable: integer grade (value param)             │
        │ Variable: integer getMessageNumber (proc. result) │
        └───────────────────────────────────────────────────┘
            ┌──────────────────────────┐
            │ level: 3, tableNumber: 3 │
            ├──────────────────────────┤
            │ Variable: real a         │
            └──────────────────────────┘
        ┌────────────────────────────────────┐
        │ level: 2, tableNumber: 4           │
        ├────────────────────────────────────┤
        │ Array: string array messages[?:3]  │
        │ Variable: string msg               │
        └────────────────────────────────────┘

Algol60> Semantic analysis successful
